### PR TITLE
fix(ui5-toolbar-*): ensure getDomRef() returns the actual DOM reference

### PR DESCRIPTION
**Pull request by kgogov**
### Bug Description

The `getDomRef()` method on `ui5-toolbar-*` items was returning `undefined` instead of a valid DOM reference. This limitation hindered developers from programmatically interacting with the toolbar items within the `ui5-toolbar` component.

### Fix Details

- Introduced a custom getDomRef() implementation tailored for ui5-toolbar items, ensuring that the method reliably returns the actual DOM reference of each rendered toolbar item.

### Outcome

Developers can now confidently access and interact with each toolbar item’s DOM element through the `getDomRef()` method, improving the overall developer experience and ensuring consistent functionality within applications.

Fixes: #9996
Related: #9997, #10126

### CODE REVIEW

File: packages/main/src/ToolbarButton.ts [hunk 1]
```diff
@@ -156,6 +156,13 @@ class ToolbarButton extends ToolbarItem {
 	@property()
 	width?: string;
 
+	onEnterDOM() {
+		const toolbar = this.parentElement as HTMLElement;
+		this._getRealDomRef = () => {
```
Comment (nnaydenow) [inline packages/main/src/ToolbarButton.ts:161]: It is not best practice to look for parent element and the context where the element is placed.

Could you please check how it is fixed in shellbar: https://github.com/SAP/ui5-webcomponents/blob/main/packages/fiori/src/ShellBar.ts#L901

Comment (kgogov) [reply packages/main/src/ToolbarButton.ts]: Thank you for the feedback! You’re absolutely right; relying on the parent element and its context isn’t ideal.

File: packages/main/test/specs/Toolbar.spec.js [hunk 2]
```diff
@@ -73,5 +73,15 @@ describe("Toolbar general interaction", () => {
 		await input.setAttribute("value", "0");
 	});
 
+	it("Should not return null upon calling getDomRef for all direct child items", async () => {
```
Comment (dobrinyonkov) [inline packages/main/test/specs/Toolbar.spec.js:76]: If possible, let's write this new test in cypress

Comment (kgogov) [reply packages/main/test/specs/Toolbar.spec.js]: Done, no problem at all!

File: packages/main/src/Toolbar.ts [hunk 3]
```diff
@@ -333,6 +333,8 @@ class Toolbar extends UI5Element {
 	onBeforeRendering() {
 		this.detachListeners();
 		this.attachListeners();
+
+		this.preprocessItems();
```
Comment (plamenivanov91) [inline packages/main/src/Toolbar.ts:337]: If I comment this line, the tests still pass. Can we secure it with tests?

**Review by plamenivanov91 — CHANGES_REQUESTED** [review]

File: packages/main/src/Toolbar.ts (hunk 3 above)
Comment (kgogov) [reply packages/main/src/Toolbar.ts:337]: Thank you for your feedback; good catch ;)

**Review by plamenivanov91 — APPROVED** [review]

**Pull request merged**